src/GraphBLAS-sharp.Backend/Common/ClArray.fs

```diff
@@ -437,6 +437,91 @@ module ClArray =
 
                 Some result
 
+    let assignWithKeysOption (op: Expr<'a -> 'b option>) (clContext: ClContext) workGroupSize =
```

Comment:
I don't quite understand what this kernel is for. Isn't it possible to express this through `mapOption` for values and `scatter` for keys? If we still want to reduce the number of basic operations...